doc/install/install.rst

```diff
@@ -0,0 +1,333 @@
+4. Python3 v3.9+ and ``pip`` (required by PETSc and OpenSn)
+5. Git version control system
+6. CMake v3.12+
+7. MPI (OpenMPI, MPICH, and MVAPICH have been tested)
+8. HDF5
+9. Doxygen (required for generating the OpenSn documentation)
+
+.. tab-set::
+
+   .. tab-item:: Linux (Ubuntu)
+
+      .. code-block:: shell
+
+         sudo apt install build-essential gfortran python3-dev python3-pip  \
+             git cmake libopenmpi-dev flex bison libhdf5-mpi-dev doxygen
+         export NPROC=$(nproc)
+
+   .. tab-item:: macOS
+
+      .. code-block:: shell
+
+         brew install gcc python git cmake open-mpi flex bison hdf5-mpi doxygen
+         export NPROC=$(sysctl -n hw.ncpu)
+
+Install PETSc
+-------------
+
+The current supported version is `PETSc version 3.23.0+ <https://web.cels.anl.gov/projects/petsc/download/release-snapshots/petsc-3.23.0.tar.gz>`_.
+
+In a scratch directory, install **PETSc** using the following commands:
+
+.. code-block:: shell
+
+   wget https://web.cels.anl.gov/projects/petsc/download/release-snapshots/petsc-3.23.0.tar.gz
+   tar -xvzf petsc-3.23.0.tar.gz
+   cd petsc-3.23.0
+
+To configure **PETSc** with the required build options, run:
+
+.. code-block:: shell
+
+   ./configure  \
+       --prefix=/path/to/dependencies/directory  \
+       --download-hypre=1  \
+       --with-ssl=0 --with-debugging=0  \
+       --with-pic=1 --with-fc=0 --with-shared-libraries=1  \
+       --download-f2cblaslapack=1  \
+       --download-metis=1 --download-parmetis=1  \
+       --download-superlu_dist=1  \
+       --download-ptscotch=1  \
+       --with-cxx-dialect=C++11  \
+       --with-64-bit-indices \
+       CC=mpicc CXX=mpicxx FC=mpifort \
+       COPTFLAGS='-O3'  \
+       CXXOPTFLAGS='-O3'  \
+       FOPTFLAGS='-O3'  \
+       PETSC_DIR=$PWD
+
+If the configuration fails, consult **PETSc**'s user documentation.
+
+Follow the **PETSc** build prompts to complete the **PETSc** installation.
+
+After a successful install, export these environment variables every time a new
+terminal is opened for launching applications with PETSc:
+
+.. code-block:: shell
+
+   export PETSC_ROOT="/path/to/dependencies/directory"
+   export CMAKE_PREFIX_PATH="${PETSC_ROOT}/lib${CMAKE_PREFIX_PATH:+:${CMAKE_PREFIX_PATH}}"
+
+Install VTK
+-----------
+
+Download `VTK 9.3.0 <https://www.vtk.org/files/release/9.3/VTK-9.3.0.tar.gz>`_ into a
+suitable location.
+
+Run the following to configure, build, and install:
+
+.. code-block:: shell
+
+   mkdir build && cd build
+   cmake \
+       -DCMAKE_INSTALL_PREFIX=/path/to/dependencies/directory  \
+       -DBUILD_SHARED_LIBS=ON  \
+       -DVTK_USE_MPI=ON  \
+       -DVTK_GROUP_ENABLE_StandAlone=WANT  \
+       -DVTK_GROUP_ENABLE_Rendering=DONT_WANT  \
+       -DVTK_GROUP_ENABLE_Imaging=DONT_WANT  \
+       -DVTK_GROUP_ENABLE_Web=DONT_WANT  \
+       -DVTK_GROUP_ENABLE_Qt=DONT_WANT  \
+       -DVTK_MODULE_USE_EXTERNAL_VTK_hdf5=ON  \
+	     -DCMAKE_BUILD_TYPE=Release  \
```

Comment:
My editor shows me: [tab][space][space][space][space][space], this should be 7 spaces